pulsar-metadata/src/test/java/org/apache/pulsar/metadata/ZKSessionTest.java

```diff
@@ -171,10 +168,12 @@ public void testReacquireLeadershipAfterSessionLost() throws Exception {
         e = sessionEvents.poll(10, TimeUnit.SECONDS);
         assertEquals(e, SessionEvent.SessionLost);
 
-        Awaitility.await().untilAsserted(() -> {
-            assertEquals(le1.getState(), LeaderElectionState.Leading);
-        });
-
+        // due to zookeeper async notice, we need to wait.
+        Awaitility.await()
+                .until(() -> le1.getState() == LeaderElectionState.Leading);
+        assertEquals(le1.getState(), LeaderElectionState.Leading);
+        Awaitility.await()
+                .until(()-> leaderElectionEvents.poll() == null);
         les = leaderElectionEvents.poll();
```

Comment:
this assertion is redundant because we are waiting for the assertion to be verified in the Awaitility loop

Reply:
Ok, I will fix it, thanks to you ~ :)